backend/routes_chemicals.py

```
@@ -241,3 +445,201 @@ def chemical_detail_route(id):
+    @app.route('/api/chemicals/<int:id>/archive', methods=['POST'])
+    @materials_manager_required
+    def archive_chemical_route(id):
+        try:
+            # Get the chemical
+            chemical = Chemical.query.get_or_404(id)
+
+            # Check if the chemical is already archived
+            try:
+                if chemical.is_archived:
+                    return jsonify({'error': 'Chemical is already archived'}), 400
+            except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)